ibis/file/csv/tests/test_basic.py

```diff
@@ -0,0 +1,116 @@
+import pytest
+
+from pandas.util import testing as tm
+import ibis
+from ibis.file.csv.client import CSVClient, CSVTable
+from ibis.file.client import Database, execute
+
+
+@pytest.fixture
+def transformed(csv):
+
+    # we need to cast to a timestamp type
+    # as we read in as strings
+    closes = csv.csv.close
+    closes = closes.mutate(time=closes.time.cast('timestamp'))
+    opens = csv.csv.open
+    opens = opens.mutate(time=opens.time.cast('timestamp'))
+
+    t = opens.inner_join(closes, ['time', 'ticker'])
+    t = t[opens, closes.close]
+    t = t.mutate(avg=(t.open + t.close) / 2)
+    t = t[['time', 'ticker', 'avg']]
+    return t
+
+
+def test_client(tmpdir, data):
+
+    # construct with a path to a file
+    csv = tmpdir
+
+    for k, v in data.items():
+        f = csv / '{}.csv'.format(k)
+        v.to_csv(str(f), index=False)
+
+    c = CSVClient(csv / 'open.csv')
+    assert c.list_databases() == []
+    assert c.list_tables() == ['open']
+
+    c = CSVClient(csv / 'close.csv')
+    assert c.list_databases() == []
+    assert c.list_tables() == ['close']
+
+
+def test_navigation(csv):
+
+    # directory navigation
+    assert isinstance(csv, Database)
+    result = dir(csv)
+    assert result == ['csv']
+
+    prices = csv.csv
+    assert isinstance(prices, Database)
+    result = dir(prices)
+    assert result == ['close', 'open']
```

Comment:
Hm, not sure about this ... I like the idea of restricting, but this seems pretty restrictive. Maybe that's okay.